### DISCUSSION TITLE
Assigning Volume ID's

### DISCUSSION THREAD

**Discussion by lonnietc** [seaweedfs/seaweedfs#2411] [Q&A]
Hello,

As I know that you can start many Volume nodes:

Start Volume Servers
> weed volume -dir="/tmp/data1" -max=5  -mserver="localhost:9333" -port=8080 &
> weed volume -dir="/tmp/data2" -max=10 -mserver="localhost:9333" -port=8081 &

and query the assign command to get the volume to write to:

> curl http://localhost:9333/dir/assign
{"count":1,"fid":"3,01637037d6","url":"127.0.0.1:8080","publicUrl":"localhost:8080"}

I am wondering if when starting up each volume, can I assign a Volume ID on the CLI such that when I call the "/dir/assign" command with Curl, then I can write to a specific volume that I have already assigned and ID?

Or, can I query a volume to have it return just the ID that it was assigned when the server was started?

These will be useful when I need to have specific files stored on specific volumes.

Any thoughts on this?

**Comment by chrislusf** [comment]
(Moving to discussion)

There are no checking when writing to volumes. So you can write to any volume id, any file key, and any cookie, as long as you can be sure about the volume having enough capacity.

One minor thing is that if the file key is not monotonically increasing, changing volume servers to use leveldb index, i.e.,`weed volume -index=leveldb`, can be more efficient.
